ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java

```diff
@@ -717,4 +812,77 @@ public void shouldSetShowDescriptionsForShowQueriesDescriptions() {
     ListQueries listQueries = (ListQueries)statement;
     Assert.assertThat(listQueries.getShowExtended(), is(true));
   }
+
+  @Test
+  public void shouldAddPrefixEvenIfColumnNameIsTheSameAsStream() {
+    final String statementString =
+        "CREATE STREAM S AS SELECT address FROM address a;";
+    final Statement statement = KSQL_PARSER.buildAst(statementString, metaStore).get(0);
+    assertThat(statement, instanceOf(CreateStreamAsSelect.class));
+    Query query = ((CreateStreamAsSelect) statement).getQuery();
+    assertThat(query.getQueryBody(), instanceOf(QuerySpecification.class));
+    QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
+    assertThat(querySpecification.getSelect().getSelectItems().get(0).toString(),
+        equalTo("A.ADDRESS ADDRESS"));
+  }
+
+  @Test
+  public void shouldNotAddPrefixIfStreamNameIsPrefix() {
+    final String statementString =
+        "CREATE STREAM S AS SELECT address.city FROM address a;";
```

Comment:
This test may be less confusing if you use a field thats in the orders schema instead. This is technically not a correct statement. Its juts that the parser isn't responsible for catching what's wrong with it.